File: reverse-string/reverseString.js
function reverseString(s) {
  for (let i = 0, j = s.length - 1; i < j; i++, j--) {
    const x = s[i];
    const y = s[j];
    s[i] = y;
    s[j] = x;
  }
}

reverseString(["h", "e", "l", "l", "o"]);

File: reverse-string/reverseString.py
def reverseString(arr):
    i = 0
    j = len(arr) - 1
    while j > i:
        tmp = arr[i]
        arr[i] = arr[j]
        arr[j] = tmp
        i += 1
        j -= 1
    return arr

    # for i in range(len(arr)):
    #     j = (len(arr) - i - 1)
    #     if i > j:
    #         break
    #     tmp = arr[i]
    #     arr[i] = arr[j]
    #     arr[j] = tmp
    # return arr


print(reverseString(["h", "e", "l", "l", "o"]))
